### PR TITLE
feat: firebase_analytics plugin with AnalyticsWithoutAdIdSupport pod

### PR DESCRIPTION
**Pull request by softkot**
As a part of Analytics 7.11.0 SDK there is a special analytics pod to follow AppTrackingTransparency. As mentioned at https://github.com/FirebaseExtended/flutterfire/discussions/6030 there is no solution to use this pod in a flutter. 

This pull request involve the new FirebaseAnalyticsWithoutAdIdSupport variable and it can be used to ovveride the default pod the same way as FirebaseSDKVersion used to [override sdk version](https://firebase.flutter.dev/docs/overview/#iosmacos)

Just add following line somewere in the head of your ios/Podfile  and  got the flutter app with AnalyticsWithoutAdIdSupport

`$FirebaseAnalyticsWithoutAdIdSupport = true`

Closes #6030

### CODE REVIEW

**Comment by Salakar** [comment]
I would prefer if this was done the same way we've done on React Native Firebase; https://github.com/invertase/react-native-firebase/pull/5224 - A flag to switch to without Ad ID Support, rather than having to specify a full string.

Would you mind matching this change to how it's ben done on here: https://github.com/invertase/react-native-firebase/pull/5224

**Comment by softkot** [comment]
@Salakar ok change it to bool $FirebaseAnalyticsWithoutAdIdSupport

**Comment by Salakar** [comment]
> @Salakar ok change it to bool $FirebaseAnalyticsWithoutAdIdSupport

Of course

**Comment by Salakar** [comment]
> @Salakar ok change it to bool $FirebaseAnalyticsWithoutAdIdSupport

Thank you! cc @mikehardy to make sure we've not missed anything here. Some documentation for this would be nice in the readme perhaps of `firebase_analytics/firebase_analytics`

**Comment by mikehardy** [comment]
This seems correct, the proof is a podfile.lock with it set both ways as a test, should get the results expected based on my read of it

**Review by Dobbe — APPROVED** [review]

File: packages/firebase_analytics/firebase_analytics/README.md [hunk 1]
```diff
@@ -9,6 +9,14 @@ For Flutter plugins for other Firebase products, see [README.md](https://github.
 ## Usage
 To use this plugin, add `firebase_analytics` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/). You must also configure firebase analytics for each platform project: Android and iOS (see the example folder for details).
 
+## iOS [App Tracking Transparency](https://developer.apple.com/documentation/apptrackingtransparency)
+
+In order to avoid any user tracking you should add following line into ios/Podfile
```
Comment (russellwheatley) [inline packages/firebase_analytics/firebase_analytics/README.md:14]:
```suggestion
If you wish to use Firebase Analytics without IDFA collection capability, open your `/ios/Podfile` and add the following global variable to the top of the file:
```

File: packages/firebase_analytics/firebase_analytics/README.md [hunk 2]
```diff
@@ -9,6 +9,14 @@ For Flutter plugins for other Firebase products, see [README.md](https://github.
...
+
+```
```
Comment (russellwheatley) [inline packages/firebase_analytics/firebase_analytics/README.md:16]:
```suggestion
```ruby
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

**Review by mikehardy — APPROVED** [review]
LGTM and matches how RNFB does it, which results in the expected difference in Podfile.lock. Would still be nice to see the result of a `cat Podfile.lock |grep AdId` to verify.

Should look like 

```
mike@bistromath:~/work/Kullki/ksocialscore/packages/public-app (p2p-features *) % cat ios/Podfile.lock |grep AdId
  - Firebase/AnalyticsWithoutAdIdSupport (8.1.1):
    - FirebaseAnalytics/WithoutAdIdSupport (~> 8.0.0)
  - FirebaseAnalytics/WithoutAdIdSupport (8.0.0):
    - GoogleAppMeasurement/WithoutAdIdSupport (= 8.0.0)
  - GoogleAppMeasurement/WithoutAdIdSupport (8.0.0):
    - Firebase/AnalyticsWithoutAdIdSupport (= 8.1.1)

```

**Review by russellwheatley — APPROVED** [review]
LGTM

**Comment by russellwheatley** [comment]
I decided to confirm to get it over the line. Here's a before (with the `$FirebaseAnalyticsWithoutAdIdSupport = true` flag) and after grep of `Podfile.lock`:
 
Screenshot 2021-07-01 at 09 33 51

**Comment by kphilemon** [comment]
Hi, may I know when this will be released? I'd need this to pass the App Store review. 
I believe editing `Podfile.lock` directly is not the correct way to do so, right?

Thanks in advance!

**Comment by softkot** [comment]
> Hi, may I know when this will be released? I'd need this to pass the App Store review.
> I believe editing `Podfile.lock` directly is not the correct way to do so, right?
> 
> Thanks in advance!

While it is not released you can override the dependency in pubspec.yaml from PR branch 

```
  firebase_analytics:
    git:
      url: https://github.com/softkot/flutterfire.git
      path: packages/firebase_analytics/firebase_analytics

```

**Comment by masewo** [comment]
Seems not to work when also trying to use firebase_crashlytics in the same project.
Got this build failure: 'FirebaseCrashlytics/FIRAnalyticsInterop.h' file not found

**Comment by mikehardy** [comment]
Interesting - I wonder if that is related to versioning - I use crashlytics and analytics-without-ad-id in a non-flutter project and the underlying pods work in that configuration. There may be something unclean in your local environment @masewo with regard to pod cache or build cache or similar

**Comment by masewo** [comment]
@mikehardy I tried following cleaning steps but still get this error (by the way: I am building on a M1):
`rm -rf Pods
rm -rf Podfile.lock
rm -rf ~/.pub-cache/hosted/pub.dartlang.org/
rm -rf ~/Library/Developer/Xcode/DerivedData
pod cache clean --all
flutter pub cache repair
flutter clean
flutter pub get
pod repo update`

**Comment by Dobbe** [comment]
@masewo I ran into the same problem with firebase_crashlytics 2.1.0. Didn't dig deeper into the problem but it might also be a problem with the Crashlytics library as I managed to get around it by downgrading the Crashlytics library to 2.0.7. 

If it seems to be a problem with the Crashlytics library, it might be worth it to file a bug there.

**Pull request merged**